api/envoy/extensions/filters/http/api_key_auth/v3/api_key_auth.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.api_key_auth.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "udpa/annotations/sensitive.proto";
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.api_key_auth.v3";
+option java_outer_classname = "ApiKeyAuthProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/api_key_auth/v3;api_key_authv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: APIKey Auth]
+// APIKey Auth :ref:`configuration overview <config_http_filters_api_key_auth>`.
+// [#extension: envoy.filters.http.api_key_auth]
+
+// Basic HTTP authentication.
+//
+// Example:
+//
+// .. code-block:: yaml
+//
+//    location: HEADER
+//    name: X-API-KEY
+//    keys:
+//      inline_string: |-
+//        apikey1:user1
+//        apikey2:user2
+//
+message APIKeyAuth {
+  enum APIKeyLocation {
+    // The API key is expected to be in a header.
+    HEADER = 0;
+
+    // The API key is expected to be in a query parameter.
+    QUERY = 1;
+
+    // The API key is expected to be in a cookie.
+    COOKIE = 2;
+  }
+
+  // The location of the API key.
+  APIKeyLocation location = 1;
+
+  // The name of the header, query parameter, or cookie that contains the API key.
+  string name = 2;
+
+  // keys used to authenticate the user.
+  // It should be a map of apikey to user.
+  config.core.v3.DataSource keys = 3 [(udpa.annotations.sensitive) = true];
```

Comment:
Do we need a `user` for API key?

Reply:
If we just want to allow/deny the requests based on keys, we don't need a concept of "user" though,
I have the use case like the rate-limiting etc based on who has the key in my mind. And, that is why I have `forward_username_header` https://github.com/envoyproxy/envoy/pull/36709#discussion_r1808638527.

Reply:
That makes sense. I prefer using `ClientID` over `User` since the owner of an API key typically is an application, not a user.